app/models/host_report.rb

```diff
@@ -55,4 +55,53 @@ def self.search_by_keyword(_key, operator, value)
       conditions: sanitize_sql_for_conditions(["host_reports.report_keyword_ids @> ?", "{#{keyword_ids.join(',')}}"]),
     }
   end
+
+  def self.report
+    @report = {} if @report.nil?
+    @report.update(
+      :total_hosts => Host.all.count,
+      :change_hosts => report_changes.count,
+      :nochange_hosts => report_nochanges.count,
+      :failure_hosts => report_failures.count,
+      :disabled_hosts => disabled_alerts.count,
+      :reports_missing => report_missing.count
+    )
+  end
+
+  def self.report_missing
+    Host.search_for('not has last_report and status.enabled = true')
+  end
+
+  def self.disabled_alerts
+    hosts.where(:enabled => true)
+  end
+
+  def self.report_changes
+    hosts.search_for("report_changes = 1")
+  end
+
+  def self.report_nochanges
+    hosts.search_for("report_nochanges = 1")
+  end
+
+  def self.report_failures
+    hosts.search_for("report_failures = 1")
+  end
+
+  def self.latest_events?
+    latest_events.limit(1).present?
+  end
+
+  def self.hosts
+    filter = nil # TODO: implement global filtering
+    hosts || Host.authorized(:view_hosts, Host).search_for(filter)
+  end
```

Comment:
What this method is supposed to return? I don't think it should be in this model, host report has exactly one host. Just merge it with the `latest_events` method it is short enough.

Reply:
@Ron-Lavi Fixed it, I'll push is soon

Reply:
I think the hosts function is needed. 
We are searching through the relevant Hosts (with the right permission and filter) which hosts are corresponded to the scoped search - "report_failures", "report_changes" and so on..
correct me if I am wrong, without that function I'll need to write these lines at every helper method

Reply:
What I was trying to say is this method alone will not use much use, so I would simply move/merge it with the `latest_events`.